drivers/gpio/gpio_max22190.c

```diff
@@ -0,0 +1,651 @@
+ * @param val - Updated value to be written in the register at update.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+static int max22190_reg_update(const struct device *dev, uint8_t addr, uint8_t mask, uint8_t val)
+{
+	int ret;
+	uint32_t reg_val = 0;
+
+	ret = max22190_reg_read(dev, addr);
+
+	reg_val = ret;
+	reg_val &= ~mask;
+	reg_val |= mask & val;
+
+	return max22190_reg_write(dev, addr, reg_val);
+}
+
+/*
+ * @brief Check FAULT1 and FAULT2
+ *
+ * @param dev - MAX22190 device
+ */
+static void max22190_fault_check(const struct device *dev)
+{
+	struct max22190_data *data = dev->data;
+
+	/* FAULT1 */
+	data->fault1.reg_raw = max22190_reg_read(dev, MAX22190_FAULT1_REG);
+
+	if (data->fault1.reg_raw) {
+		/* FAULT1_EN */
+		data->fault1_en.reg_raw = max22190_reg_read(dev, MAX22190_FAULT1_EN_REG);
+
+		PRINT_ERR_BIT(data->fault1.reg_bits.max22190_CRC,
+				data->fault1_en.reg_bits.max22190_CRCE);
+		PRINT_ERR_BIT(data->fault1.reg_bits.max22190_POR,
+				data->fault1_en.reg_bits.max22190_PORE);
+		PRINT_ERR_BIT(data->fault1.reg_bits.max22190_FAULT2,
+				data->fault1_en.reg_bits.max22190_FAULT2E);
+		PRINT_ERR_BIT(data->fault1.reg_bits.max22190_ALRMT2,
+				data->fault1_en.reg_bits.max22190_ALRMT2E);
+		PRINT_ERR_BIT(data->fault1.reg_bits.max22190_ALRMT1,
+				data->fault1_en.reg_bits.max22190_ALRMT1E);
+		PRINT_ERR_BIT(data->fault1.reg_bits.max22190_24VL,
+				data->fault1_en.reg_bits.max22190_24VLE);
+		PRINT_ERR_BIT(data->fault1.reg_bits.max22190_24VM,
+				data->fault1_en.reg_bits.max22190_24VME);
+		PRINT_ERR_BIT(data->fault1.reg_bits.max22190_WBG,
+				data->fault1_en.reg_bits.max22190_WBGE);
+
+		if (data->fault1.reg_bits.max22190_WBG & data->fault1_en.reg_bits.max22190_WBGE) {
+			uint8_t wb_val = max22190_reg_read(dev, MAX22190_WB_REG);
+
+			max22190_update_wb_stat(dev, (uint8_t)wb_val);
+		}
+
+		if (data->fault1.reg_bits.max22190_FAULT2) {
+			/* FAULT2 */
+			data->fault2.reg_raw = max22190_reg_read(dev, MAX22190_FAULT2_REG);
+
+			/* FAULT2_EN */
+			data->fault2_en.reg_raw = max22190_reg_read(dev, MAX22190_FAULT2_EN_REG);
+
+			PRINT_ERR_BIT(data->fault2.reg_bits.max22190_RFWBS,
+					data->fault2_en.reg_bits.max22190_RFWBSE);
+			PRINT_ERR_BIT(data->fault2.reg_bits.max22190_RFWBO,
+					data->fault2_en.reg_bits.max22190_RFWBOE);
+			PRINT_ERR_BIT(data->fault2.reg_bits.max22190_RFDIS,
+					data->fault2_en.reg_bits.max22190_RFDISE);
+			PRINT_ERR_BIT(data->fault2.reg_bits.max22190_RFDIO,
+					data->fault2_en.reg_bits.max22190_RFDIOE);
+			PRINT_ERR_BIT(data->fault2.reg_bits.max22190_OTSHDN,
+					data->fault2_en.reg_bits.max22190_OTSHDNE);
+			PRINT_ERR_BIT(data->fault2.reg_bits.max22190_FAULT8CK,
+					data->fault2_en.reg_bits.max22190_FAULT8CKE);
+		}
+	}
+}
+
+static void max22190_state_get(const struct device *dev)
+{
+	const struct max22190_config *config = dev->config;
+
+	if (gpio_pin_get_dt(&config->fault_gpio))
+		max22190_fault_check(dev);
```

Comment:
Add braces to every `if`, `else`, `do`, `while`, `for` and `switch` body, even for single-line code blocks
https://docs.zephyrproject.org/latest/contribute/guidelines.html#coding-style

Reply:
Done